npm/design-system/src/util/types.ts

```diff
@@ -1 +1 @@
-export type ExtractFirstArg<T extends (...args: any[]) => unknown> = T extends (arg0: infer S, ...otherArgs: any[]) => unknown ? S : never
+export type ExtractFirstArg<T extends (...args: any[]) => unknown> = Parameters<T>[0]
```

Comment:
Could you add a quick comment on this type and how to use it?
I consider myself well versed in typing challenges and it took me a hot minute to figure out.